tests/components/apns/test_notify.py

```diff
@@ -23,356 +23,367 @@
+    """Test setup with missing topic."""
+    config = {
+        "notify": {
+            "platform": "apns",
+            "name": "test_app",
+            "cert_file": "test_app.pem",
         }
-        with assert_setup_component(0) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert not handle_config[notify.DOMAIN]
-
-    @patch("homeassistant.components.apns.notify._write_device")
-    def test_register_new_device(self, mock_write):
-        """Test registering a new device with a name."""
-        yaml_file = {5678: {"name": "test device 2"}}
-
-        written_devices = []
-
-        def fake_write(_out, device):
-            """Fake write_device."""
-            written_devices.append(device)
-
-        mock_write.side_effect = fake_write
-
-        with patch(
-            "homeassistant.components.apns.notify.load_yaml_config_file",
-            Mock(return_value=yaml_file),
-        ):
-            self._setup_notify()
-
-        assert self.hass.services.call(
-            apns.DOMAIN,
-            "apns_test_app",
-            {"push_id": "1234", "name": "test device"},
-            blocking=True,
-        )
+    }
+    with assert_setup_component(0) as handle_config:
+        assert await async_setup_component(hass, notify.DOMAIN, config)
+    assert not handle_config[notify.DOMAIN]
 
-        assert len(written_devices) == 1
-        assert written_devices[0].name == "test device"
 
-    @patch("homeassistant.components.apns.notify._write_device")
-    def test_register_device_without_name(self, mock_write):
-        """Test registering a without a name."""
-        yaml_file = {
-            1234: {"name": "test device 1", "tracking_device_id": "tracking123"},
-            5678: {"name": "test device 2", "tracking_device_id": "tracking456"},
-        }
+@patch("homeassistant.components.apns.notify._write_device")
+async def test_register_new_device(mock_write, hass):
+    """Test registering a new device with a name."""
+    yaml_file = {5678: {"name": "test device 2"}}
 
-        written_devices = []
+    written_devices = []
 
-        def fake_write(_out, device):
-            """Fake write_device."""
-            written_devices.append(device)
+    def fake_write(_out, device):
+        """Fake write_device."""
+        written_devices.append(device)
 
-        mock_write.side_effect = fake_write
+    mock_write.side_effect = fake_write
 
-        with patch(
-            "homeassistant.components.apns.notify.load_yaml_config_file",
-            Mock(return_value=yaml_file),
-        ):
-            self._setup_notify()
+    with patch(
+        "homeassistant.components.apns.notify.load_yaml_config_file",
+        Mock(return_value=yaml_file),
+    ):
+        await _setup_notify(hass)
 
-        assert self.hass.services.call(
-            apns.DOMAIN, "apns_test_app", {"push_id": "1234"}, blocking=True
-        )
+    assert await hass.services.async_call(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)